pf/tests/schemashim_test.go

```diff
@@ -152,15 +152,10 @@ func TestSchemaShimRepresentations(t *testing.T) {
     "_": {
       "list_attribute": {
         "element": {
-          "schema": {
-            "element": {
-              "resource": {
-                "a1": {
-                  "type": 4
-                }
-              }
-            },
-            "type": 6
```

Comment:
type=6 is TypeMap I think. This looks like the test was expecting what Elem() describes as "single-nested block encoding" - "In this case s.Type() == TypeMap and s.Elem() is a Resource value."

IN the new encoding, this is encoded as TypeList with Elem()=Resource. Elem() doc comment describes that this is an encoding for List-nested block but with this change it also encodes a List<Obj> attribute.

Reply:
It the old encoding it is List(Elem(Map(Elem(Resource))))

In the new one it is List(Elem(Resource))

Given that pulumi doesn't care about attribute vs block, we might want to encode list nested blocks and list nested objects the same. Both represent an object in a list as far as pulumi is concerned.

Reply:
Also in the old encoding `list_nested_attribute` and `list with object elem` are encoded the same. I'm not sure I understand why these would be different to `list_nested_block` in the shim layer.

Reply:
In the sdkv2 list nested attributes are also represented the same as list nested blocks: https://github.com/pulumi/pulumi-terraform-bridge/pull/2242

Reply:
> Given that pulumi doesn't care about attribute vs block

Are we 100% sure. I was trying to remember if that's the case, can't find anything just now searching now. There is normalizeBlockCollections but it's scoped to sdkv2 "under" the Shim abstraction.